dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/response/IResponseImpl.java

```diff
@@ -59,51 +59,71 @@ public IResponseImpl(IRequest req, HttpInboundConnection connection)
     this.allocateDirect = false;
 
   }  
-  
+ 
+  /*
+   * Since Servlet 6.0 - refactor to support Cookie setAttribute
+   */
+  protected HttpCookie addCookieHelper(Cookie cookie) {
+      String methodName = "addCookieHelper";
+      String cookieName = cookie.getName();
+      
+      if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())  {
+          Tr.entry(tc, methodName + " cookie ["+cookieName+"]");
+      }
+
+      HttpCookie hc = new HttpCookie(cookieName, cookie.getValue());
+      hc.setPath(cookie.getPath());
+      hc.setVersion(cookie.getVersion());
+      hc.setComment(cookie.getComment());
+      hc.setDomain(cookie.getDomain());
+      hc.setMaxAge(cookie.getMaxAge());
+      hc.setSecure(cookie.getSecure());
+      hc.setHttpOnly(cookie.isHttpOnly());
+
+      /*
+       * Check to see if the WebContainerRequestState has an attribute defined for 
+       * the Cookie that is being added.
+       *
+       * If the attribute is not recognized by the Channel Framework then it is ignored.
```

Comment:
The change being made here adds all the attributes as the cookie being parsed, so the channel framework does not ignore them after this change, right? Perhaps this comment is a bit misleading since the behavior is changing.

Reply:
Instead of removing it, append the message at the end to make it clear.

       * If the attribute is not recognized by the Channel Framework then it is ignored **(in Servlet 5.0 and earlier)**